src/QsCompiler/Core/Transformations/SyntaxTreeTransformation.fs

```diff
@@ -48,270 +45,218 @@ type SyntaxTreeTransformation<'T> private (state: 'T, options: TransformationOpt
 
     member this.SharedState = state
 
-    new(state: 'T, options: TransformationOptions) as this =
-        SyntaxTreeTransformation<'T>(state, options, "_internal_")
-        then
-            this.Types <- new TypeTransformation<'T>(this, options)
-            this.ExpressionKinds <- new ExpressionKindTransformation<'T>(this, options)
-            this.Expressions <- new ExpressionTransformation<'T>(this, options)
-            this.StatementKinds <- new StatementKindTransformation<'T>(this, options)
-            this.Statements <- new StatementTransformation<'T>(this, options)
-            this.Namespaces <- new NamespaceTransformation<'T>(this, options)
-
-    new(state: 'T) = new SyntaxTreeTransformation<'T>(state, TransformationOptions.Default)
+    new(state) = SyntaxTreeTransformation(state, TransformationOptions.Default)
 
-    // These overrides are here since it is impractical to have them live in the definition of CommonTransformationItems.
+    abstract OnLocalNameDeclaration: name: string -> string
```

Comment:
Why is `OnLocalNameDeclaration` not set to be replaced with the identity function in 2022-09 if it is implemented in the identity function?

Reply:
Maybe the comment phrasing is confusing - what I mean is that

```fsharp
default this.OnLocalName name = this.Statements.OnVariableName name
```

should be replaced by

```fsharp
default this.OnLocalName name = name
```

OnLocalNameDeclaration already has this kind of implementation (because it doesn't have a backwards compatibility shim like OnLocalName), so a todo isn't needed.